File: host/lessons/00-rust-orientation/comparisons/threshold.py
def is_above_limit(reading: int, limit: int) -> bool:
    return reading > limit


def main() -> None:
    readings = [18, 22, 25, 29]
    limit = 25
    qualifying_count = 0

    for reading in readings:
        if is_above_limit(reading, limit):
            qualifying_count += 1

    print(f"Readings above {limit}: {qualifying_count}")


if __name__ == "__main__":
    main()

File: host/lessons/00-rust-orientation/comparisons/threshold.cpp
#include <array>
#include <iostream>

bool is_above_limit(int reading, int limit) {
    return reading > limit;
}

int main() {
    const std::array<int, 4> readings = {18, 22, 25, 29};
    const int limit = 25;
    int qualifying_count = 0;

    for (int reading : readings) {
        if (is_above_limit(reading, limit)) {
            qualifying_count += 1;
        }
    }

    std::cout << "Readings above " << limit << ": " << qualifying_count << '\n';
}
